velox/exec/Operator.cpp

```diff
@@ -415,14 +417,17 @@ void Operator::MemoryReclaimer::enterArbitration() {
   // The driver must be alive as the operator is still under memory arbitration
   // processing.
   VELOX_CHECK_NOT_NULL(driver);
-  if (FOLLY_UNLIKELY(!driver->state().isOnThread())) {
+  if (FOLLY_UNLIKELY(
+          !driver->state().isOnThread() ||
+          (std::this_thread::get_id() != driver->state().thread))) {
```

Comment:
If driver is on thread, but this call happens on a non-driver thread, do we not need to do anything? Would you clarify?

Reply:
Updated. Thanks!